src/platforms/wayland/display.cpp

```diff
@@ -269,15 +276,28 @@ try
         {
             wl_display_cancel_read(display);
         }
+
+        if (fds[roundtrip].revents & (POLLIN | POLLERR))
+        {
+            eventfd_t foo;
+            eventfd_read(roundtrip_signal, &foo);
+            wl_display_roundtrip(display);
+        }
     }
 }
 catch (std::exception const& e)
 {
     fatal_error("Critical error in Wayland platform: %s\n", boost::diagnostic_information(e).c_str());
 }
 
+void mir::graphics::wayland::Display::roundtrip() const
+{
+    eventfd_write(roundtrip_signal, 1);
```

Comment:
Does this block until the roundtrip has been completed? How? If not, what's the point of calling it at all?

Reply:
No. That happens on the Wayland thread.

The point is that this empties the queue of message to the server. (Which needs to happen on the Wayland thread.) Is there a better way?

Reply:
/me reads the documentation and finds "wl_display_flush()"